src/base/settings/run.rs

```diff
@@ -60,6 +63,30 @@ pub struct Config {
     pub cache: Option<bool>,
 }
 
+#[derive(Deserialize, Clone, Debug)]
+#[allow(non_snake_case)]
+pub struct Advanced {
+    #[serde(default)]
+    pub THETA_E: f64, // Convergence criteria
+    #[serde(default)]
+    pub THETA_G: f64, // Objective function stop criteria
+    #[serde(default)]
+    pub THETA_F: f64, // ?
+    #[serde(default)]
+    pub THETA_D: f64, // ?
+}
+
+impl Default for Advanced {
+    fn default() -> Self {
+        Self {
+            THETA_E: 1e-4,
+            THETA_G: 1e-4,
+            THETA_F: 1e-2,
+            THETA_D: 1e-2,
```

Comment:
This default should be 1e-4, no?

Reply:
These were copied from the current code in npag.rs - should all of them be 1e-4?

Reply:
const THETA_E: f64 = 1e-4; //convergence Criteria
const THETA_G: f64 = 1e-4; //objf stop criteria
const THETA_F: f64 = 1e-2;
const THETA_D: f64 = 1e-4;

This is what we have in the code right now

Reply:
You are absolutely right - my bad! I'll correct them.